src/reaction_network.jl

```diff
@@ -57,287 +57,119 @@ See the [Chemical Reaction Model
-macro reaction_network(ex::Expr, p...)
-    coordinate(:reaction_network, MacroTools.striplines(ex), p, :no___noise___scaling)
-end
-
-################# query-based macros:
-
-"""
-    @min_reaction_network
-
-Generates a subtype of an `AbstractReactionNetwork` that only encodes a chemical
-reaction network. Use [`addodes!`](@ref), [`addsdes!`](@ref) or
-[`addjumps!`](@ref) to complete the network for specific problem types. It accepts
-the same arguments as [`@reaction_network`](@ref).
-"""
-macro min_reaction_network(name, ex::Expr, p...)
-    min_coordinate(name, MacroTools.striplines(ex), p, :no___noise___scaling)
-end
 
-#Macro to create a reaction network model. Multiple dispatch is used to allow for SDE noise scalling.
-macro min_reaction_network(name, scale_noise, ex::Expr, p...)
-    in(scale_noise, p) || (p = (p..., scale_noise))
-    min_coordinate(name, MacroTools.striplines(ex), p, scale_noise)
-end
-
-#If no type name is given, creates a network with a default name.
-macro min_reaction_network(ex::Expr, p...)
-    min_coordinate(:min_reaction_network, MacroTools.striplines(ex), p, :no___noise___scaling)
-end
-
-"""
-    @empty_reaction_network networktype
-
-Generates a subtype of an `AbstractReactionNetwork` that encodes an empty
-chemical reaction network. `networktype` is an optional parameter that specifies
-the type of the generated network. Use [`addspecies!`](@ref), [`addparam!`](@ref)
-and [`addreaction!`](@ref) to extend the network.  Use [`addodes!`](@ref),
-[`addsdes!`](@ref) or [`addjumps!`](@ref) to complete the network for specific
-problem types.
-"""
-macro empty_reaction_network(name::Symbol=:min_reaction_network)
-    min_coordinate(name, MacroTools.striplines(:(begin end)), (), :no___noise___scaling)
-end
-
-#################
-
-#Declare various arrow types symbols used for the empty set (also 0).
+# Declare various arrow types symbols used for the empty set (also 0).
 empty_set = Set{Symbol}([:∅])
 fwd_arrows = Set{Symbol}([:>, :→, :↣, :↦, :⇾, :⟶, :⟼, :⥟, :⥟, :⇀, :⇁, :⇒, :⟾])
 bwd_arrows = Set{Symbol}([:<, :←, :↢, :↤, :⇽, :⟵, :⟻, :⥚, :⥞, :↼, :↽, :⇐, :⟽])
 double_arrows = Set{Symbol}([:↔, :⟷, :⇄, :⇆, :⇔, :⟺])
-no_mass_arrows = Set{Symbol}([:⇐, :⟽, :⇒, :⟾, :⇔, :⟺])      #Using this arrows will disable the program from multiplying reaction rates with the substrate concentrations. Gives user full control of reaction rates.
-
-funcdict = Dict{Symbol, Function}()                             #Stores user defined functions.
-
-#Coordination function, actually does all the work of the macro.
-function coordinate(name, ex::Expr, p, scale_noise)
+pure_rate_arrows = Set{Symbol}([:⇐, :⟽, :⇒, :⟾, :⇔, :⟺])
 
-    # minimal reaction network components
-    (reactions, reactants, parameters, syms, params) = get_minnetwork(ex, p)
+# Declares symbols which may neither be used as paraemters not varriables.
+forbidden_symbols = [:t, :π, :pi, :ℯ, :im, :I, :nothing, :∅]
```

Comment:
Maybe we should allow `I`? It is very common in SIR models (for example), which are already in much of the documentation.

Reply:
Good point. I just had some memory of this giving weird errors back in the beginning and added it to the lsit. But you're right, if there's no error to it then given its prominence in SIR (and the docs) we should allow it.

Reply:
It will error if the `LinearAlgebra` package is loaded. What functionality from it are we using in the main package now? Can we just import what's needed?

Reply:
If you want to leave this for now I can get it when I take over... You've already done plenty!

Reply:
Don't know. Trying to remove it right now, to check what happens.

Reply:
Seems fine actually. Likely we can just remove `LinearAlgebra`.